src/install/lockfile.zig

```diff
@@ -3470,9 +3471,17 @@ pub const Package = extern struct {
                 list: for (keys, version_strings, 0..) |key, version_string_, i| {
                     // Duplicate peer & dev dependencies are promoted to whichever appeared first
                     // In practice, npm validates this so it shouldn't happen
-                    if (comptime group.behavior.isPeer() or group.behavior.isDev()) {
-                        for (dependencies[0..total_dependencies_count]) |dependency| {
-                            if (dependency.name_hash == key.hash) continue :list;
+                    var duplicate_at: ?usize = null;
+                    if (comptime group.behavior.isPeer() or group.behavior.isDev() or group.behavior.isOptional()) {
```

Comment:
I think we only need this check if it isOptional right? Because the inner for loop only breaks if isOptional

Reply:
We want peer and dev dependencies to enter the loop as well, because they will hit `continue :list;`. It would probably be more clear like `if (!group.behavior.isNormal()) { ... }`